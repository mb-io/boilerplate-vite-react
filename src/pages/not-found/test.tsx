import { render } from '@testing-library/react'
import NotFoundPage from './not-found'
import { describe, it, expect } from 'vitest'
import { BrowserRouter } from 'react-router-dom'

describe('NotFoundPage', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <NotFoundPage />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
