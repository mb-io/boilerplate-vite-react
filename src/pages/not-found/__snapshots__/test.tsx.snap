// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > renders correctly 1`] = `
<div>
  <section
    class="flex flex-col items-center justify-center gap-4 py-8 md:py-10"
  >
    <div
      class="flex items-center justify-center"
    >
      <div
        class="flex size-12 items-center justify-center rounded-full bg-primary-50"
      >
        <div
          class="flex size-5 items-center justify-center rounded-full border-1 border-primary"
        >
          <svg
            class="text-primary"
            fill="currentColor"
            height="12"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 128 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 64c0-17.7-14.3-32-32-32S32 46.3 32 64l0 256c0 17.7 14.3 32 32 32s32-14.3 32-32L96 64zM64 480a40 40 0 1 0 0-80 40 40 0 1 0 0 80z"
            />
          </svg>
        </div>
      </div>
    </div>
    <h1
      class="inline font-semibold tracking-tight text-[2.3rem] leading-9 lg:text-5xl"
    >
      Page not found
    </h1>
    <p>
      The page you are looking for doesn't exist.
    </p>
    <div
      class="flex items-center justify-center gap-4"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 288 480 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-370.7 0 73.4-73.4c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-128 128z"
          />
        </svg>
        Go back
      </button>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
        type="button"
      >
        Take me home
      </button>
    </div>
  </section>
</div>
`;
